src/emqx_acl_cache.erl

```diff
@@ -239,8 +255,19 @@ time_now() -> erlang:system_time(millisecond).
 if_expired(CachedAt, Fun) ->
     TTL = get_cache_ttl(),
     Now = time_now(),
-    if (CachedAt + TTL) =< Now ->
-           Fun(true);
-       true ->
-           Fun(false)
+    case (CachedAt + TTL) =< Now of
```

Comment:
Disclaimer: I am not familiar with the code. But looking at the diff, perhaps a simpler implementation is possible, that doesn't require `is_discarded`. Instead, change implementation of `if_expired` to:

```erlang
case CachedAt =< EvictionTimestamp andalso (CachedAt + TTL) =< Now of
...
```
where `EvictionTimestamp` is obtained from the persistent term. It contains current time instead of a counter. There's a downside, though, that currently this module doesn't use monotonic time, so in theory it will be possible that node timewarps into the past, and starts dropping all caches.

Reply:
`EvictionTimestamp` serves as a unique marker, rather not used as an actual timestamp.

The idea is to set the stamp from any process,
then all the connection process can check if the stamp has been changed since they last time.
if yes, discard all cache.

Reply:
I mean the timestamp compare maybe works OK. but it's confusing.

Reply:
Actually @k32 might be right. if we changed `EvictionTimestamp` role from unique marker to a timestamp, then (checking in `if_expired`) all cache entries added before `EvictionTimestamp` would become automagically evicted. 
Up to you @zmstone which way we go, im just learning here :-)

Reply:
A second thought, since is_expired is down in the stack (except for dump_acl_cache which does not respect TTL anyway), we can avoid a PD put to remember the last stamp.

@sh3d2, please change it as @k32 suggested.